monitor-e2e/test/nativeToErc.js

```diff
@@ -14,3 +15,19 @@ describe('NATIVE TO ERC', () => {
   it('balance', () => assert(parseInt(data.home.balance, 10) >= 0))
   it('should contain totalSupply', () => assert(data.foreign.totalSupply === '0'))
 })
+
+describe('NATIVE TO ERC with changing state of contracts', () => {
+  let data
+
+  before(async () => {
+    ;({ data } = await axios.get(`${baseUrl}`))
+    assert(data.balanceDiff === 0)
+  })
+
+  it('should change balanceDiff', async () => {
+    await sendEther(homeRPC.URL, user, nativeToErcBridge.home)
+    checkAll()
```

Comment:
could you clarify the logic of the test more? I don't understand it.

Reply:
1. The precondition is that `balanceDiff === 0`
2. `sendEther` is used to send ether to only one side of the bridge
3. New data is fetched from the monitor
    - `checkAll()` so that the monitor runs it's `checkWorker` scripts
    - `axios.get` so that we get new data from the monitor
4. Expected result is that `balanceDiff !== 0`

Reply:
OK. Understood. This is due to the fact that our monitor currently is not re-generating data on periodical basis. In actual deployment we use `cron` for this purposes.
But supporting docker-in-docker environment just to work this limitation around is overhead for me.
I see several possible options that could help us:
1. Implement a new entry point in https://github.com/poanetwork/tokenbridge/blob/more-monitor-e2e-tests/monitor/index.js as so it will re-generate the data when receives the GET request. For security reasons this entry point could be available only in the testing environment. In this case the GET request can be sent after the `sendEther` call. 
2. Implement a new target for testing environment that will call a script that will re-generate the data every 2-4 seconds. So, instead of calling `yarn check-and-start`
https://github.com/poanetwork/tokenbridge/blob/1f611704414e5d40c7c71050a34b263d991b743c/e2e-commons/docker-compose.yml#L242
    it can call this script. 
3. Similarly to `wait-for-monitor.sh` in https://github.com/poanetwork/tokenbridge/blob/more-monitor-e2e-tests/monitor-e2e/run-tests.sh we can introduce a script that will re-generate the data on periodical manner in background. 

What do you think? I see that usage docker-in-docker approach is more straightforward but necessity to install docker only for these needs makes it unviable for me. Especially due to the issue you described here: https://github.com/poanetwork/tokenbridge/pull/191/files#r316196312

Reply:
I agree, and every option you proposed makes sense to me.
At first sight option `3.` seems most natural to me.
I will look into it more deeply and try to implement it.